port/CMakeLists.txt

```diff
@@ -178,9 +178,15 @@ if (GDAL_USE_ICONV)
   if (Iconv_IS_BUILT_IN)
     target_sources(cpl PRIVATE cpl_recode_iconv.cpp)
   else ()
-    if (APPLE)
-      target_compile_definitions(cpl PRIVATE -DLIBICONV_PLUG)
-    endif ()
+      if (APPLE)
+          # linkage against OSX's system iconv is going to require
+          # LIBICONV_PLUG to be defined, but other self-built situations
+          # like CondaForge or Homebrew might not need this
+          check_symbol_exists(_iconv_close "iconv.h" HAVE_ICONV_CLOSE)
```

Comment:
how does it know in which library to look for the symbol ? I would expect ``set(CMAKE_REQUIRED_LIBRARIES ${Iconv_LIBRARY})`` to be set before